service/internal/graph/graph.go

```diff
@@ -288,7 +288,9 @@ func (g *Graph) buildComponents(ctx context.Context, set Settings) error {
 		case *connectorNode:
 			err = n.buildComponent(ctx, telemetrySettings, set.BuildInfo, set.ConnectorBuilder, g.nextConsumers(n.ID()))
 		case *capabilitiesNode:
-			capability := consumer.Capabilities{MutatesData: false}
+			capability := consumer.Capabilities{
+				MutatesData: g.pipelines[n.pipelineID].fanOutNode.getConsumer().Capabilities().MutatesData,
```

Comment:
Is the exporter fanout? If that is the case please comment in the code or rename the variable to clarify it.

Reply:
The fanOutNode always comes after processors and before exporters, so should represent the aggregate capabilities of the exporter in the pipeline. I'll add a comment.

Reply:
I've added a comment.